core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -96,35 +97,44 @@ trait MethodConfigurationComponent {
-
-        (methodConfigurationPrereqQuery ++= prerequisites) andThen
-          (methodConfigurationInputQuery ++= inputs) andThen
-            (methodConfigurationOutputQuery ++= outputs)
-      }
+    // the save method is used for both creating a new method config and updating an edited method config
+    // the save method is not used for renaming a method config, see the rename method for that
+    def save(workspaceContext: SlickWorkspaceContext, methodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
 
       uniqueResult[MethodConfigurationRecord](findByName(workspaceContext.workspaceId, methodConfig.namespace, methodConfig.name)) flatMap {
         case None =>
-          val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, methodConfig))
-          configInsert flatMap { configId =>
-            saveMaps(configId)
+            val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, methodConfig))
+            configInsert flatMap { configId =>
+              saveMaps(methodConfig, configId)
           }
         case Some(methodConfigRec) =>
-          workspaceQuery.updateLastModified(workspaceContext.workspaceId) andThen
-            findInputsByConfigId(methodConfigRec.id).delete andThen
-            findOutputsByConfigId(methodConfigRec.id).delete andThen
-            findPrereqsByConfigId(methodConfigRec.id).delete andThen
-            findById(methodConfigRec.id).map(rec => (rec.methodNamespace, rec.methodName, rec.methodVersion)).update(methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion) andThen
-            saveMaps(methodConfigRec.id) andThen
-            findByName(workspaceContext.workspaceId, methodConfig.namespace, methodConfig.name).map(_.rootEntityType).update(methodConfig.rootEntityType)
+          saveNewVersion(workspaceContext, methodConfigRec.id, methodConfigRec.name, methodConfigRec.methodConfigVersion, methodConfig)
       }
     } map { _ => methodConfig }
 
+    private def saveMaps(methodConfig: MethodConfiguration, configId: Long) = {
+      val prerequisites = methodConfig.prerequisites.map { case (key, value) => marshalConfigPrereq(configId, key, value) }
+      val inputs = methodConfig.inputs.map { case (key, value) => marshalConfigInput(configId, key, value) }
+      val outputs = methodConfig.outputs.map{ case (key, value) => marshalConfigOutput(configId, key, value) }
+
+      (methodConfigurationPrereqQuery ++= prerequisites) andThen
+        (methodConfigurationInputQuery ++= inputs) andThen
+        (methodConfigurationOutputQuery ++= outputs)
+    }
+
+    private def saveNewVersion(workspaceContext: SlickWorkspaceContext, oldId: Long, oldName: String, oldConfigVersion: Int, newMethodConfig: MethodConfiguration) = {
```

Comment:
I find this function confusing -- I think you need to make it clearer what "old" and "new" mean here. I think "old" means "current", right? You might want to pass in the entire `methodConfigRec` (above) representing the current-version to save on parameters and make it a bit clearer. I think this method could do with a comment to explain what it does: "hides the current record and make a new record, incrementing the version from the current". You should also make it clear that this function expects them to share a name and namespace, since the function makes absolutely no sense if this isn't true.